### PR TITLE
feat: add in-app notification, refactor toast

### LINKED ISSUE #14296: Implement In-app Notifications component
Figma: https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System?node-id=1686%3A34385

### PR DESCRIPTION
**Pull request by yqrashawn**
fixes #14296

  * [x] refactor toast component
  * [x] add in-app notification component
  * [x] add in-app notification preview
  * [x] component test

reuse the same `:toasts/upsert` event

``` clojure
(rf/dispatch [:toasts/upsert
              {...
               :type :notification
               ...}])
```

status: wip

### CODE REVIEW

File: src/quo2/components/notifications/toast/view.cljs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+   [blur/view
+    {:style         style/blur-container
+     :blur-amount   13
+     :blur-radius   10
+     :blur-type     :transparent
+     :overlay-color :transparent}]
+   [rn/view
+    {:style (style/content-container override-theme)}
+    [rn/view {:style {:padding 2}} left]
+    [rn/view {:style {:padding 4 :flex 1}}
```
Comment (J-Son89) [inline src/quo2/components/notifications/toast/view.cljs]: move to style file 🙏

File: src/quo2/core.cljs [hunk 2]
```diff
@@ -48,7 +49,9 @@
     quo2.components.notifications.count-down-circle
     quo2.components.notifications.info-count
     quo2.components.notifications.notification-dot
-    quo2.components.notifications.toast
+    quo2.components.notifications.notification.view
+    quo2.components.notifications.toast.view
+    quo2.components.onboarding.small-option-card.view
```
Comment (J-Son89) [inline src/quo2/core.cljs:55]: 🌟

File: src/quo2/core.cljs [hunk 3]
```diff
@@ -62,20 +65,19 @@
...
+    quo2.components.tags.status-tags
     quo2.components.tags.tag
     quo2.components.tags.tags
     quo2.components.tags.token-tag
-    quo2.components.text-combinations.title.view
-    quo2.components.list-items.user-list))
+    quo2.components.text-combinations.title.view))
 
-(def toast quo2.components.notifications.toast/toast)
+(def notification quo2.components.notifications.notification.view/notification)
```
Comment (J-Son89) [inline src/quo2/core.cljs]: can you put these in a commented section below like the rest. keeping it alpha sorted would be great too 👍 

i.e 
```
;;;; Notifications
(def notification quo2.components.notifications.notification.view/notification)
(def toast quo2.components.notifications.toast.view/toast)

```

File: src/status_im2/contexts/quo_preview/notifications/notification.cljs [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+     ]]))
+
+(defn preview-notification
+  []
+  [rn/view {:flex 1}
+   [rn/flat-list
+    {:flex                      1
+     :header                    [preview]
+     :key-fn                    str
+     :keyboardShouldPersistTaps :always}]])
```
Comment (J-Son89) [inline src/status_im2/contexts/quo_preview/notifications/notification.cljs]: can you make this key kebab-case?

Comment (J-Son89) [reply src/status_im2/contexts/quo_preview/notifications/notification.cljs]: sorry for review comments, I see now it's still WIP 😅

**Comment by yqrashawn** [comment]
@J-Son89 fixed above comments

**Review by J-Son89 — APPROVED** [review]

**Comment by J-Son89** [comment]
nice work with the refactors too :D

**Pull request merged**